### PR TITLE
Update UI

### PR DESCRIPTION
**Pull request by muppi1993**
UI has been restructured:
After Logging in, a home page allows the user to select between running new analysis, or seeing previous results/ uploaded data (not implemented yet). Navigation is also available through a dropdown menu in the header, which also provides the logout option. The new Project view displays the dropzone as well as initial web worker results, and changes to a tab view after hitting the submit button. So far, only 'table' and 'network' tab are available, this can be extended in the future by the SKA phylogeny tree tab.

### CODE REVIEW

File: app/client/src/components/DropZone.vue [hunk 1]
```diff
@@ -2,8 +2,9 @@
   <div>
     <div v-bind='getRootProps()' class=dropzone>
       <input v-bind='getInputProps()' />
-      <p v-if='isDragActive'>Drop the files here ...</p>
-      <p v-else>Drag and drop your fasta files here, or click to select files</p>
+      <p v-if='isDragActive' class="dropzone-text">Drop the files here ...</p>
+      <p v-else class="dropzone-text">Drag and drop your fasta files here,
+        or click to select files</p>
```
Comment (EmmaLRussell) [inline app/client/src/components/DropZone.vue:7]: I guess this change of text is the reason for the width change on drag over, should just need some css to make fixed width.

**Review by EmmaLRussell — COMMENTED** [review]
This works really well! 👍 

Got a few comments on UI details. Since Marie is leaving, the following comments are not requests for her to make changes in this PR - most could be done in separate branches (some of them might already have tickets?):
- I see a strange behaviour in Firefox when dragging files onto the drop area - the width momentarily shrinks when the file is dragged over, and expands again after it is dropped. 
- We're going to put in 'Analysis history' next, but probably not 'Manage my data' for a while, so we should probably hide that button on the Home page to start with
- The big empty space on centre and right of home page seems a bit odd - was this originally intended to have some content? Might feel more natural to centre the buttons? Or maybe we should combine the Analysis history and Home pages?
- The file table could start off with a 'no files uploaded yet' message rather than just being empty
- The visual relationship between 'Create a new project' header and 'Start Analysis' button looks a bit odd
- The names of the buttons on the homepage and the headings of the pages they link to should be consistent e.g. currently 'Run new analysis' links to 'Create a new project' - should have consistent terminology (which of 'project' or 'analysis' would make more sense to user?)
- It might be nice to have an animation to slide away the file drop area when the analysis starts - it's a little bit jarring to have it suddenly disappear
- It might be nice to have the cluster numbers in the table as links which can be clicked to open up the cluster on the Cluster tab. 
- I think we'll need to sort out starting a new project as an already logged in user vs returning to my current project. Given the name of the button, we'd probably expect that 'Run new analysis' would start a fresh project rather than return to an existing one - but would still want to be able to return to current too. But we can sort that one when we put in project history and management.

**Comment by EmmaLRussell** [comment]
Tagging @richfitz - could you check if you agree with my comments, and see if you have any others? I'll make the smaller changes here, and create tickets for the rest.

File: app/client/src/components/SelectAction.vue [hunk 2]
```diff
@@ -0,0 +1,37 @@
+<template>
+  <div class="select-action">
+    <i class="bi bi-house-fill left house"></i>
+    <p class="left">Welcome back, {{user.name}}!</p>
```
[the remark this answers is not in the document]
Comment (richfitz) [reply app/client/src/components/SelectAction.vue:4]: sounds like something user persistance would help us with :stuck_out_tongue_closed_eyes:

**Review by richfitz — COMMENTED** [review]
* scrollbars on startup are a bit odd
* probably best not to display table or "start analysis" button until files uploaded?
* it might be more accessible to also have an upload file button too? I've seen this done nicely where when you drag files towards an input it becomes a drop zone
* it's quite alarming when the input shrinks away from the input as if flinching - that might be worth fixing before the merge?
* The "Create a new project" title when the analysis is running is probably not wanted - at that point we have a project?
* The "Start analysis" button should not be displayed while analysis is running, or after it has completed.
* The progress bar probably should go away once everything is all done, perhaps replace with some summary (like the "Completed 3 of 3 jobs" that is already displayed).
* The front page should have some text about why we need the user to login, and also a privacy/data use statement that can be read before login
* In the dropdown menu, the item is only activated when clicking the text, not anywhere in the row, which is confusing

Focused on the app not the code, to get something in quickly

**Comment by EmmaLRussell** [comment]
I've made an epic for adapting the UI to a more MINTy/WODINy style - starting with some mockups, and including some of the comments above.: https://mrc-ide.myjetbrains.com/youtrack/issue/bacpop-91/Beebop-UI

In this branch, I've made the following changes to address the remainder of the comments:
- avoid drop area width change on drag over
- hide 'Manage my data' button
- 'No files uploaded yet' message on empty table
- some simplifications to template code
- avoid vertical scrollbar on Home page
- drop down menu items should be fully clickable

**Comment by richfitz** [comment]
Looking good - I can still see scrollbars though, at any level of zoom:

image

**Comment by EmmaLRussell** [comment]
> Looking good - I can still see scrollbars though, at any level of zoom:
> 
Ah interesting, I was only seeing the vertical scrollbar for the page, not on the inner component like that. Is that on Chrome?

**Comment by richfitz** [comment]
yeah, this is Chrome

**Comment by EmmaLRussell** [comment]
> yeah, this is Chrome

Try it now?

**Comment by richfitz** [comment]
image

Looking good now on the scrollbars - I do see this odd alignment issue on the project page though, which I think is new?

**Comment by EmmaLRussell** [comment]
> Looking good now on the scrollbars - I do see this odd alignment issue on the project page though, which I think is new?

I don't think it's new, thought maybe it looks a little different with the table change - I'd already made the comment

> The visual relationship between 'Create a new project' header and 'Start Analysis' button looks a bit odd

..which I've included in the list of things to address in the mockups. Could try tidying it up in this branch if you prefer?

**Review by richfitz — APPROVED** [review]
Sorry, losing track of what we were dealing with at which point. Let's close this out before I get more confused

**Pull request merged**